src/main/java/org/snomed/snowstorm/core/data/services/ConceptService.java

```diff
@@ -153,55 +158,62 @@ public Concept find(String id, List<LanguageDialect> languageDialects, BranchTim
 	}
 
 	public ConceptHistory loadConceptHistory(String conceptId, List<CodeSystemVersion> codeSystemVersions, boolean showFutureVersions) {
+		int now = Integer.parseInt(DATE_TIME_FORMATTER.format(Instant.now()));
 		Map<String, BranchCriteria> branchCriteria = new HashMap<>();
 		for (CodeSystemVersion codeSystemVersion : codeSystemVersions) {
-			String branchPath = codeSystemVersion.getBranchPath();
-			branchCriteria.put(branchPath, versionControlHelper.getBranchCriteria(branchPath));
+			if (showFutureVersions || codeSystemVersion.getEffectiveDate() < now) {
```

Comment:
As suggested, I am now only adding a branchCriteria to my map if showFutureVersions is indeed true or if the codeSystemVersion is not in the future; this will prevent the logic from being carried out directly on ES.

Reply:
The call to `codeSystemService.findAllVersions(codeSystem.getShortName(), showFutureVersions)` in the concept controller should already take care of the filtering. You shouldn't need to check the version effectiveDate here. You may not need `showFutureVersions` as a parameter of this method (`loadConceptHistory`).

Reply:
Fixed.